url/src/main/java/com/datastax/oss/dsbulk/url/S3URLStreamHandler.java

```diff
@@ -81,11 +85,11 @@ public InputStream getInputStream() {
       GetObjectRequest getObjectRequest =
           GetObjectRequest.builder().bucket(bucket).key(key).build();
       String query = url.getQuery();
-      if (query == null) {
+      if (StringUtils.isBlank(query)) {
         throw new IllegalArgumentException(
             "You must provide S3 client credentials in the URL query parameters.");
       }
-      S3Client s3Client = s3ClientCache.computeIfAbsent(query, this::getS3Client);
+      S3Client s3Client = s3ClientCache.get(query, this::getS3Client);
```

Comment:
By using a raw string as the cache key, we would create 2 distinct clients for semantically equivalent queries, e.g. `foo=0&bar=1` and `bar=1&foo=0`. Wouldn't it be better to sanitize the string, or use something more structured as the cache key, maybe a `S3ClientInfo` class?